engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/MetricsProviderForFlink.scala

```diff
@@ -1,18 +1,18 @@
 package pl.touk.nussknacker.engine.process.compiler
 
 import org.apache.flink.api.common.functions.RuntimeContext
-import pl.touk.nussknacker.engine.ComponentUseCase
+import pl.touk.nussknacker.engine.ComponentUseContextProvider
 import pl.touk.nussknacker.engine.util.metrics.{MetricsProviderForScenario, NoOpMetricsProviderForScenario}
 
 object MetricsProviderForFlink {
 
   def createMetricsProvider(
-      componentUseCase: ComponentUseCase,
+      componentUseContextProvider: ComponentUseContextProvider,
       runtimeContext: RuntimeContext
   ): MetricsProviderForScenario = {
-    componentUseCase match {
-      case ComponentUseCase.TestRuntime => NoOpMetricsProviderForScenario
-      case _                            => new FlinkMetricsProviderForScenario(runtimeContext)
+    componentUseContextProvider match {
```

Comment:
Hmm I see that we use this class not only to provide `ComponentUseContext` but also for some pattern matching for other mechanism. I think that we should revert the name to sth like `RuntimeMode` or sth like that.

Reply:
I like the name